### ISSUE TITLE
Recursion max exceeded error

### ISSUE THREAD

**Issue by ipelupessy** [amusecode/amuse#123]
a recent commit, I think  commit  614c2e5791700578269bf609098809e0d985611d can trigger a condition where __del__ of a code leads to a recursion limit error, eg
```
touch src/amuse/rfi/core.py
```
and then in a python shell
```
from amuse.community.sse.interface import SSE
s=SSE() # leads to warning about code as expected
```
then exit with ctrl-d
gives:
```
Exception RuntimeError: 'maximum recursion depth exceeded while calling a Python object' in <bound method SSE.__del__ of <amuse.community.sse.interface.SSE object at 0x6c1f90>> ignored
```

**Comment by ipelupessy** [comment]
the test case above triggeres the error, but there is some weirdness involved: it does not happen for all codes, and it may relate to the fact that self.stop() doesn't exist (hasattr(self,"stop") returns false), but adding a check doesn't help. in fact replacing the stop with a print statement makes the error go away, but merely checking some attribute triggers the same error...

**Comment by arjenve** [comment]
it seems it fails whenever the code is in a incorrect state, this could easily happen when the code is ready for clean-up, I removed the __del__ method

**Comment by ipelupessy** [comment]
fixed by 46cfee3bd1464673dd883cbdbe9f7902c044ab96
